ui/components/app/transaction-status-label/transaction-status-label.test.js

```diff
@@ -1,103 +1,81 @@
-      },
-      selectedAccount: TEST_ACCOUNT_ID,
-    },
-  },
-});
+// Mock the Tooltip component
+jest.mock('../../ui/tooltip', () => ({
+  __esModule: true,
+  default: ({ children }) => <div data-testid="tooltip">{children}</div>,
+}));
 
-const statusTestCases = [
-  {
-    name: 'CONFIRMED',
-    props: { status: 'confirmed', date: 'June 1' },
-  },
-  {
-    name: 'PENDING',
-    props: {
+describe('TransactionStatusLabel Component', () => {
+  it('should render CONFIRMED status and date', () => {
+    const props = {
+      status: 'confirmed',
       date: 'June 1',
+      statusOnly: false,
+    };
+
+    render(<TransactionStatusLabel {...props} />);
+    expect(screen.getByText('June 1')).toBeInTheDocument();
+  });
+
+  it('should render PENDING status when submitted and isEarliestNonce is true', () => {
+    const props = {
       status: TransactionStatus.submitted,
       isEarliestNonce: true,
-    },
-  },
-  {
-    name: 'QUEUED',
-    props: {
+    };
+
+    render(<TransactionStatusLabel {...props} />);
+    expect(
+      screen.getByText(TransactionGroupStatus.pending),
+    ).toBeInTheDocument();
+  });
+
+  it('should render QUEUED status when submitted and isEarliestNonce is false', () => {
+    const props = {
       status: TransactionStatus.submitted,
       isEarliestNonce: false,
-    },
-  },
-  {
-    name: 'UNAPPROVED',
-    props: {
-      status: TransactionStatus.unapproved,
-    },
-  },
-  {
-    name: 'SIGNING',
-    props: {
-      status: TransactionStatus.approved,
-    },
-  },
-];
+    };
 
-describe('TransactionStatusLabel Component', () => {
-  const createMockStore = configureMockStore([thunk]);
-  let store;
+    render(<TransactionStatusLabel {...props} />);
+    expect(screen.getByText('queued')).toBeInTheDocument();
+  });
 
-  beforeEach(() => {
-    const basicAccount = createBasicAccount();
-    const mockState = createMockStateWithAccount(basicAccount);
-    store = createMockStore(mockState);
+  it('should render UNAPPROVED status', () => {
+    const props = {
+      status: TransactionStatus.unapproved,
+    };
+
+    render(<TransactionStatusLabel {...props} />);
+    expect(screen.getByText(TransactionStatus.unapproved)).toBeInTheDocument();
   });
 
-  statusTestCases.forEach(({ name, props }) => {
-    it(`renders ${name} properly and tooltip`, () => {
-      const { container, queryByTestId } = renderWithProvider(
-        <TransactionStatusLabel {...props} />,
-        store,
-      );
-      expect(container).toMatchSnapshot();
-      expect(queryByTestId('transaction-status-label')).not.toBeInTheDocument();
-    });
+  it('should render SIGNING status when approved', () => {
+    const props = {
+      status: TransactionStatus.approved,
+    };
+
+    render(<TransactionStatusLabel {...props} />);
+    expect(screen.getByText('signing')).toBeInTheDocument();
   });
 
-  it('renders pure text for status when shouldShowTooltip is specified as false', () => {
```

Comment:
we're missing the general tooltip tests that were previously there before the reverted commit and with the reverted commit

https://github.com/MetaMask/metamask-extension/pull/29338/files#diff-3b95bca7eb6f7377dc2165dcec7d2545fbf8c4437a37432040d68279703dffa3L119

I think it might be good to re-add here so we don't accidentally lose this one

Reply:
@digiwand, good catch. I've added tests to check for the status.  Please take a look when you have a moment.

Reply:
Re: 'renders pure text for status when shouldShowTooltip is specified as false'

Removing this test since we're no longer using the shouldShowTooltip feature (which was added in Danica's PR but is now being reverted).